src/components/select-custom/select-custom.component.ts

```diff
@@ -0,0 +1,128 @@
+import { ChangeDetectionStrategy, Component, ContentChild, ElementRef, EventEmitter, forwardRef, Input, OnChanges, OnDestroy, OnInit, Output, SimpleChanges, TemplateRef, ViewChild } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'ux-select-custom',
+    templateUrl: './select-custom.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            multi: true,
+            useExisting: forwardRef(() => SelectCustomComponent)
+        }
+    ],
+    host: {
+        '[style.width.px]': 'width'
+    }
+})
+export class SelectCustomComponent<T> implements ControlValueAccessor, OnChanges, OnInit, OnDestroy {
+
+    /** Define the selected item */
+    @Input() selected: T;
```

Comment:
This is due to the "base" / "custom" nature of this component: The selection is always performed in the contained component. We include these components in reactive forms and therefore, they need to implement ControlValueAccessor and provide the selected element.

Reply:
May want to compare with the way the [Date/Time Picker](https://pages.github.houston.softwaregrp.net/caf/ux-aspects-micro-focus/#/ui-components/input-controls#date-time-picker) works.

The contents of the popover in this case does not need to be a form control. We simply bind the output `date` to the `ngModel` of the input field, which is this case is already usable in a form, but the same principle can apply to a custom ControlValueAccessor component.

Reply:
But that means that whatever is in the drop-down content, needs to be wrapped a separate component, so no more chance for simple implementations.

Our source CI picker looks like this at the moment:

```
<opr-ngx-select-custom formControlName="sourceCi">

  <ng-template #buttonContent>
    <span *ngIf="sourceCIControl.value && (sourceCIControl.value.name.length > 0)">
      {{ sourceCIControl.value.label }}
      <span class="ci-name">({{ sourceCIControl.value.ciTypeLabel }})</span>
    </span>
  </ng-template>

  <ng-template #dropdownContent>
    <ul class="ci-list">
      <li *ngFor="let ci of sourceCIList"
          (click)="editorForm.patchValue({sourceCi: ci})"
          [class.selected]="selected(ci,sourceCIControl.value)">
        {{ ci.ciTypeLabel }} <span class="ci-name">({{ ci.label }})</span>
      </li>
    </ul>
  </ng-template>

</opr-ngx-select-custom>
```

Such a compact implementation will then not be possible anymore.

Reply:
The example in our site does not necessarily need to have a separate component, but a separate component is a valid use case that we need to consider, in which case we may not be able to simply do a (click) event to patch the value.

The confusing thing is with the code above, the custom select is bound to a form control but doesn't actually do the updating. The list below is still the thing updating the form control value. It's not a pattern have I really seen before where the `formControlName` element is not the thing updating the form?

Reply:
The above example is a little strange. I don't think binding between parent and child should be done via FormControl.

Referring back to the date/time picker example, it would look something like this. 

````html
<opr-ngx-select-custom [(value)]="selectCustomValue" formControlName="sourceCi">
  <ng-template #dropdownContent>
    <ux-number-picker [(value)]="selectCustomValue"></ux-number-picker>
  </ng-template>
</opr-ngx-select-custom>
````

It would make no difference whether a custom component is used or not.

Reply:
I am not sure I get your point. Which change to the component are you proposing? Or does each of you propose a different solution?

Reply:
I'm proposing that you don't misuse the reactive forms feature to sync the value between child and parent. I have already pointed you to the date & time picker example, as well as including a pseudocode example, showing how it should be structured. Unless there are good reasons not to follow this model, let's go with it.

Reply:
I was asking mainly because your and Ashley's proposals seemed to contradict each other and I wanted to come to a common solution.

About your last proposal: it looks very much like my current implementation, except that you call it value and I call it selected. Is it only about renaming the attributes? Am I missing anything?

Reply:
It was a response to this: https://github.com/UXAspects/UXAspects/pull/1161#discussion_r312448433, and the ensuing discussion about whether the inner component should be the form control.

Reply:
So is there still a change requested regarding this topic, and which one, or can everything stay as it is?

Reply:
I believe that the `ng-content` recommendation by Ashley will take care of all of the above concerns.